src/test/java/htsjdk/variant/vcf/VCFHeaderUnitTest.java

```diff
@@ -503,6 +516,39 @@ public void testVCFHeaderQuoteEscaping() throws Exception {
 
     }
 
+    @Test
+    public void testVcf42Roundtrip() throws Exception {
+        // this test ensures that source/version fields are round-tripped properly
+
+        // read an existing VCF
+        File expectedFile = new File("src/test/resources/htsjdk/variant/Vcf4.2WithSourceVersionInfoFields.vcf");
+        final VCFFileReader originalFileReader = new VCFFileReader(expectedFile, false);
+        final VCFHeader originalHeader = originalFileReader.getFileHeader();
+        
+        // write the file out into a new copy
+        final File actualFile = File.createTempFile("testVcf4.2roundtrip.", IOUtil.VCF_FILE_EXTENSION);
+        actualFile.deleteOnExit();
+
+        final VariantContextWriter copyWriter = new VariantContextWriterBuilder()
+                .setOutputFile(actualFile)
+                .setReferenceDictionary(createArtificialSequenceDictionary())
+                .setOptions(EnumSet.of(Options.ALLOW_MISSING_FIELDS_IN_HEADER, Options.INDEX_ON_THE_FLY))
+                .build();
+        copyWriter.writeHeader(originalHeader);
+        final CloseableIterator<VariantContext> firstCopyVariantIterator = originalFileReader.iterator();
+        while (firstCopyVariantIterator.hasNext()) {
+            VariantContext variantContext = firstCopyVariantIterator.next();
+            copyWriter.add(variantContext);
+        }
+        originalFileReader.close();
+        copyWriter.close();
```

Comment:
Use try-with-resources for both of these to avoid manual closing.

Reply:
I copy&pasted your own code here...